spec/istring.dd

```diff
@@ -0,0 +1,228 @@
+string literal data and values. An interpolation expression is written like a
+string, but can contain values that are passed directly to a function that is
+able to accept them. It is transformed into a $(I Sequence) of expressions that
+can be overloaded or handled by template functions.)
+
+$(H2 $(LNAME2 ies, IES Literals))
+
+$(GRAMMAR
+$(GNAME InterpolationExpressionSequence):
+    $(GLINK InterpolatedDoubleQuotedLiteral)
+    $(GLINK InterpolatedWysiwygLiteral)
+    $(GLINK InterpolatedTokenLiteral)
+)
+
+$(P An Interpolation Expression sequence can either wysiwyg quoted, double
+quoted, or a token literal. Only double quoted literals can have escapes in
+them.)
+
+$(P Unlike string literals, IES literals cannot have a suffix defining the
+width of the character type for the string expressions.)
+
+$(H3 $(LNAME2 ies_doublequoted, Double Quoted IES Literals))
+
+$(GRAMMAR
+$(GNAME InterpolatedDoubleQuotedLiteral):
+    $(B i") $(GLINK InterpolatedDoubleQuotedCharacters)$(OPT) $(B ")
+
+$(GNAME InterpolatedDoubleQuotedCharacters):
+    $(GLINK InterpolatedDoubleQuotedCharacter)
+    $(GLINK InterpolatedDoubleQuotedCharacter) $(GSELF InterpolatedDoubleQuotedCharacters)
+
+$(GNAME InterpolatedDoubleQuotedCharacter):
+    $(GLINK2 lex, DoubleQuotedCharacter)
+    $(GLINK InterpolationEscapeSequence)
+    $(GLINK InterpolationExpression)
+
+$(GNAME InterpolationEscapeSequence):
+    $(GLINK EscapeSequence)
+    $(B \\\$)
+
+$(GNAME InterpolationExpression):
+    $(B \$$(LPAREN)) $(GLINK AssignExpression) $(B $(RPAREN))
+)
+
+$(P Like $(GLINK2 lex, DoubleQuotedString), double-quoted IES literals can
+have escape characters in them. Added to the normal escapes is the ability to
+escape a literal $)
+
+$(P A $ followed by any character other than a left parenthesis is treated as a
+literal $ in the expression, there is no need to escape it.)
+
+$(P The expression inside an $(GLINK InterpolationExpression) is a full D
+expression, and escapes are not needed inside that part of the expression.)
+
+$(H3 $(LNAME2 ies_wysiwyg, Wysiwyg IES Literals))
+
+$(GRAMMAR
+$(GNAME InterpolatedWysiwygLiteral):
+    $(B i$(BACKTICK)) $(GLINK InterpolatedWysiwygCharacters)$(OPT) $(B $(BACKTICK))
+
+$(GNAME InterpolatedWysiwygCharacters):
+    $(GLINK InterpolatedWysiwygCharacter)
+    $(GLINK InterpolatedWysiwygCharacter) $(GSELF InterpolatedWysiwygCharacters)
+
+$(GNAME InterpolatedWysiwygCharacter):
+    $(GLINK2 lex, WysiwygCharacter)
+    $(GLINK InterpolationExpression)
+)
+
+$(P Wysiwyg ("what you see is what you get") IES literals are defined like
+$(GLINK2 lex, WysiwygString) strings, but only support backquote syntax. No
+escapes are recognized inside these literals.)
+
+$(H3 $(LNAME2 ies_token, Token IES Literals))
+
+$(GRAMMAR
+$(GNAME InterpolatedTokenLiteral):
+    $(D iq{) $(GLINK InterpolatedTokenStringTokens)$(OPT) $(D })
+
+$(GNAME InterpolatedTokenStringTokens):
+    $(GLINK InterpolatedTokenStringTokenNoBraces)
```

Comment:
InterpolatedTokenStringTokenNoBraces is not defined. Here is a PR: https://github.com/dlang/dlang.org/pull/3772